db/database_collection.go

```diff
@@ -38,27 +41,42 @@ func (c *DatabaseCollection) AllowConflicts() bool {
 	return base.DefaultAllowConflicts
 }
 
+// allPrincipalIDs returns the IDs of all users and roles, including deleted Roles
+func (c *DatabaseCollection) allPrincipalIDs(ctx context.Context) (users, roles []string, err error) {
+	return c.dbCtx.AllPrincipalIDs(ctx)
+}
+
 // Authenticator returns authentication options associated with the collection's database.
 func (c *DatabaseCollection) Authenticator(ctx context.Context) *auth.Authenticator {
 	return c.dbCtx.Authenticator(ctx)
 }
 
+// activeChannels tracks active replications by channel. This is a database level property.
+func (c *DatabaseCollection) activeChannels() *channels.ActiveChannels {
+	return c.dbCtx.activeChannels
+}
+
 // backupOldRev creates a temporary copy of a old revision body when resolving conflicts. This is controlled at the database level.
 func (c *DatabaseCollection) backupOldRev() bool {
 	return c.dbCtx.Options.ImportOptions.BackupOldRev
 
 }
 
-// changeCache returns the set of channels in memory. This is currently at the database level.
-func (c *DatabaseCollection) changeCache() *changeCache {
-	return c.dbCtx.changeCache
+// ChangeCache returns the set of channels in memory. This is currently at the database level.
+func (c *DatabaseCollection) ChangeCache() *changeCache {
```

Comment:
Is this public just to avoid the naming collision with c.changeCache?  If so, do we really need the accessor function?

If there's a good reason for it, I think we should still avoid making this public, either by renaming to GetChangeCache(), or rename the struct object changeCache_.  (neither are great, to be honest, would prefer omitting unless there's a good reason.)

Reply:
Actually removing this, for its single use outside of DB exposed a bug where docIDs where removed from all channel caches. This is still hard to write unit tests for until we can have multiple collections. I wrote `DatabaseCollection.RemoveFromChangeCache`.